### ISSUE TITLE
Forward Passport's `info`

### ISSUE THREAD

**Issue by sroze** [nestjs/passport#24]
When using [`passport-http`](https://github.com/jaredhanson/passport-http) to provide some `Basic` authentication, I don't receive the Realm in the failure response and therefore I'm not prompted by the browser.

It is [setting the realm as the failure reason](https://github.com/jaredhanson/passport-http/blob/master/lib/passport-http/strategies/basic.js#L107). Because we ignore [the `info` argument](https://github.com/nestjs/passport/blob/master/lib/auth.guard.ts#L65) then it's never sent back to the client.

Any idea on how you'd see this working? I'm happy to do the PR.

**Comment by lutvianes** [comment]
For now, exception filter can be used to trigger `Basic` authentication prompt by browser. At least this code works for me.
```
@Catch(UnauthorizedException)
export class BasicAuthExceptionFilter extends BaseExceptionFilter<UnauthorizedException> {
    constructor(@Inject(HTTP_SERVER_REF) applicationRef?: HttpServer) {
        super(applicationRef)
    }

    catch(exception: UnauthorizedException, host: ArgumentsHost) {
        const ctx = host.switchToHttp();
        const response = ctx.getResponse();
        const request = ctx.getRequest();
        const status = exception.getStatus()

        if (request.authInfo) {
            response
                .status(status)
                .set('WWW-Authenticate', request.authInfo)
                .send()
        } else {
            super.catch(exception, host)
        }
    }
}
```

**Comment by kamilmysliwiec** [comment]
Done #29. Published as 6.1.0

**Comment by Roaders** [comment]
Many many thanks for that code snippet @lutvianes - there's really not much help available when googling why basic auth does not work in nest and your code fixed it for me. The only bit that was missing for me was hooking the filter up in my controller:

```typescript
    @Get('timeBasic')
    @UseGuards(BasicAuthGuard)
    @UseFilters(BasicAuthExceptionFilter)
    getTimeBasic() {
        return this.appService.getTime();
    }
```

And a different constructor param for the filter (I think that `HTTP_SERVER_REF` has been deprecated)

```typescript
import { BaseExceptionFilter, HttpAdapterHost } from '@nestjs/core';

@Catch(UnauthorizedException)
export class BasicAuthExceptionFilter extends BaseExceptionFilter<UnauthorizedException> {
    constructor(adapterHost: HttpAdapterHost) {
        super(adapterHost.httpAdapter);
    }
```
